app/commands/solution/unlock_help.rb

```diff
@@ -0,0 +1,11 @@
+class Solution::UnlockHelp
+  include Mandate
+
+  initialize_with :solution
+
+  def call
+    raise SolutionCannotBeUnlockedError unless solution.downloaded? || solution.submitted?
+
+    @solution.update(unlocked_help: true)
```

Comment:
```suggestion
    @solution.update!(unlocked_help: true)
```

Reply:
You might want to also consider catching this if it raises and reraising as SolutionCannotBeUnlockedError 🤷

Reply:
What type of error would `update!` throw? As in: what should I catch?